src/Components/Admin/ObjectEditor/unorderedpuzzle.js

```diff
@@ -158,18 +149,35 @@ export default function UnorderedPuzzle(props) {
         reorderImages(index, Math.min(images.length - 1, index + 1));
     };
 
+    const addImage = () => {
+        setImages([...images, {}]);
+        props.addImage();
+        setUploaded(false);
+    };
+
+    const deleteImage = (index) => {
+        const tempImages = JSON.parse(JSON.stringify(images));
+        tempImages.splice(index, 1);
+        setImages(tempImages);
+        props.deleteImage(index);
+        setUploaded(false);
+    };
+
     return (
         <div>
             {isUnordered ? (
-                <Select
-                    value={imagesLengthList.filter(
-                        (option) => option.value === images?.length
-                    )}
-                    options={imagesLengthList}
-                    placeholder="Select number of images..."
-                    searchable={false}
-                    onChange={selectPuzzleImages}
-                />
+                <div>
+                    Add Images
+                    <IconButton
+                        className={props.classes.addButton}
+                        aria-label="add"
+                        onClick={() => {
+                            addImage();
+                        }}
+                    >
+                        <AddIcon />
+                    </IconButton>
+                </div>
```

Comment:
So you know how there is a disabled prop on the delete button when # images == 2, should we have a disabled prop on add button when # images == 5?

Reply:
Good idea, I think we probably should

Reply:
Can you add that and I'll re-review

Reply:
Added